srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,298 @@
+import type { CSSResultGroup, LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+import { BslibElement } from "./webcomponents/_bslibElement";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+type ToggleMessage = {
+  method: "toggle";
+  value: DarkModeSwitch["themeValue"] | "toggle";
+};
+
+// This class extends BslibElement only as a convenience for registering the custom element
+// and Shiny input binding. All the logic here (besides the `isShinyInput`/`tagName` fields) is
+// designed to work without Bootstrap/bslib
+export class DarkModeSwitch extends BslibElement {
+  static isShinyInput = true;
+  static tagName = "bslib-dark-mode-switch";
+  private themeAttribute = "data-shinytheme";
+
+  @property({ type: String, attribute: "theme-value", reflect: true })
+  // eslint-disable-next-line indent
```

Comment:
It looks like prettier and eslint have different ideas about how the indentation should be, and that's why this comment is needed.

I suggest removing the eslint rule for indentation here:
https://github.com/rstudio/bslib/blob/512ad57e443c6b9730d801caa4c06ab1539008f1/.eslintrc.yml#L31-L34

And also have prettier format on save. This can be done with a .vscode/settings.json file, like this:
https://github.com/wch/shinycomponent/blob/f4d8080544a273d9e41287f678f4dfe0d39e4ffb/.vscode/settings.json#L11-L14

(This may be worth doing in a separate PR though)

Reply:
Yeah that was exactly the problem. Those lines are also in the [shiny `.eslintrc.yml` here](https://github.com/rstudio/shiny/blob/b054e45402ee31f1e58cb6e1d1f51f76f98a0aca/.eslintrc.yml#L33-L36). I'm not sure if there's something else bslib-specific happening, otherwise I'd rather try to address these settings globally across our repos.

Reply:
https://github.com/rstudio/shiny/issues/3888